hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/config/QuestionsApiConfig.java

```diff
@@ -28,11 +28,21 @@
 
 import static org.alfresco.hxi_connector.common.util.EnsureUtils.ensurePositive;
 
-public record QuestionsApiConfig(
-        int maxContextSizeForQuestion)
+import lombok.Data;
+
+@Data
+public class QuestionsApiConfig
 {
-    public QuestionsApiConfig
+    private int maxContextSizeForQuestion;
+
+    public QuestionsApiConfig(int maxContextSizeForQuestion)
```

Comment:
I assume this constructor (and the setter below too) is "stronger" than lombok's `@Data`?

Reply:
Lombok generates a `RequiredArgsConstructor` which in this case should be a default constructor. I overrode the setter since I wanted to preserve the validation.